### ISSUE TITLE
Phase vs frequency plots look very inconsistent between ragavi and shadems

### ISSUE THREAD

**Issue by molnard89** [caracal-pipeline/caracal#1159]
When I plot phase vs channel of the same dataset, using the same column, I get very different looking figures between ragavi and shadems. I'll put the relevant config file snippet below for both with figures attached:

```
inspect__2:
  enable: true
  label_in: 'cal'
  field: 'calibrators'
  label_plot: 'primary_only'
  dirname: crosscalPO
  standard_plotter: none
  shadems:
    enable: true
    ignore_errors: true
    default_column: CORRECTED_DATA
    plots_by_field:
      - "-x FREQ   -y phase -c ANTENNA1 --corr XX --cnum 64 --field 1"
      - "-x FREQ   -y phase -c ANTENNA1 --corr YY --cnum 64 --field 1"
```

And the resulting shadems plots:

![first-centSplit32k-primary_only-bpcal-J0408-6545-CORRECTED_DATA-XX-phase-FREQ-ANTENNA1](https://user-images.githubusercontent.com/43236491/83679617-0007f400-a5e0-11ea-87db-c5152d4c5384.png)

![first-centSplit32k-primary_only-bpcal-J0408-6545-CORRECTED_DATA-YY-phase-FREQ-ANTENNA1](https://user-images.githubusercontent.com/43236491/83679623-04341180-a5e0-11ea-86b9-b560cbc938c5.png)

Now the ragavi version:

```
inspect__3:
  enable: true
  label_in: 'cal'
  field: 'calibrators'
  label_plot: 'primary_only'
  dirname: crosscalPO
  phase_chan:
    enable: true
    avgtime: '60'
```

And the resulting plots:

<img width="1440" alt="ragavi_phase_chan_YY" src="https://user-images.githubusercontent.com/43236491/83679979-93412980-a5e0-11ea-8b4d-8d72c7f43fbc.png">

<img width="1432" alt="ragavi_phase_chan_XX" src="https://user-images.githubusercontent.com/43236491/83679993-976d4700-a5e0-11ea-8984-89fb24f3819b.png">

-400 to 400 degrees in phase in itself looks wild, so in combination with the shadems plot, we think this is likely some plotting issue, not data related. Any ideas what might be going on?

**Comment by o-smirnov** [comment]
Can has MS please?

**Comment by molnard89** [comment]
@o-smirnov it's 720 gig so... @paoloserra ?

**Comment by paoloserra** [comment]
Just the primary and just  CORRECTED would do? That's all there is in those plots.

**Comment by SpheMakh** [comment]
The behaviour seems pretty consistent across scans, maybe just split one of the scans to make it even smaller.

**Comment by SpheMakh** [comment]
@molnard89 can you also generate a plot using plotms for good measure.

**Comment by molnard89** [comment]
@SpheMakh ok will do.

It's on whiplash, so I can move it to some shared folder once I split the relevant data.

**Comment by KshitijT** [comment]
> -400 to 400 degrees in phase in itself looks wild

I thought this was already fixed? You are using the shared singularity images on Whiplash, @molnard89 ?

**Comment by molnard89** [comment]
@KshitijT yes I do, from `/software/astro/caracal/STIMELA_IMAGES_1.6.1/`

**Comment by o-smirnov** [comment]
Any news on that data?..

**Comment by molnard89** [comment]
@o-smirnov sorry had a meeting, splitting it now. Where should I move it on ilifu?

**Comment by o-smirnov** [comment]
Anywhere I can read it. I'd rather you put it on stills.ru.ac.za please.

**Comment by molnard89** [comment]
I can't ssh into stills I'm afraid cause `Permission denied (publickey)`

**Comment by o-smirnov** [comment]
Ah, can you e-mail me your public key please?

**Comment by o-smirnov** [comment]
MS subset available as ``/net/jake/home/molnar/msTest/NGC136532k_cal_scan25.ms`` now.

But I gotta say, I am unable to reproduce. What is the version of ragavi inside the Stimela image? @Mulan-94 @SpheMakh can you tell? @Mulan-94 could this be the old phase winding issue somehow? Or the wrong column being plotted? @molnard89 could you please also post the ragavi and shadems logs from the original run?

With the uploaded subset and the current pip release of ragavi, I get plots entirely consistent with shadems, and entirely consistent with perfectly calibrated data:

![image](https://user-images.githubusercontent.com/6470079/83808632-6f034c80-a6b5-11ea-8554-81d9c4a82c1c.png)

![image](https://user-images.githubusercontent.com/6470079/83808671-7b87a500-a6b5-11ea-9734-629019523e6c.png)

![image](https://user-images.githubusercontent.com/6470079/83808681-7fb3c280-a6b5-11ea-9f5d-29624bb68050.png)

It's kind of ironic @molnard89 that your original shadems plot in https://github.com/caracal-pipeline/caracal/issues/1159#issue-630262756 looks wrong, but I think it's actually perfectly right. You seem to have such a nicely uniform noise distribution, all the vivid individual colours have blended together into a uniform brownish hue. The one visualization regime where data is at its best when it looks like shit!

**Comment by molnard89** [comment]
Thanks for the checks! According to the logs it's ragavi 1.5.0 that lives in the image folder we're using on whiplash.

Log snippet from ragavi:

```
[####################################### ] | 99% Completed | 26.1s
[########################################] | 100% Completed | 26.2s
03.06.2020@17:42:59 - ragavi       - INFO       - Aggregation done
03.06.2020@17:42:59 - ragavi       - INFO       - image_callback executed in: 26.3 sec.
03.06.2020@17:42:59 - ragavi       - INFO       - Creating Bokeh Figure
03.06.2020@17:42:59 - ragavi       - INFO       - Plotting complete.
03.06.2020@17:42:59 - ragavi       - INFO       - Rendered plot to: /stimela_mount/output/first-centSplit32k-primary_only-bpcal-J0408-6545-phase_chan-CorrXX.html
03.06.2020@17:42:59 - ragavi       - INFO       - mytabs: ['/stimela_mount/msdir/centSplit32k_cal.ms'], xaxis: channel, yaxis: phase, c_height: 300, c_width: 300, mycmap: blues, n_cols: 5, iter_axis: scan,
                                                  html_name: /stimela_mount/output/first-centSplit32k-primary_only-bpcal-J0408-6545-phase_chan-CorrXX, corr: 0,
                                                  data_column: CORRECTED_DATA, fields: 1, flag: True, mem_limit: 8GB, n_cores: 8
03.06.2020@17:42:59 - ragavi       - INFO       - >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
Running ragavi-vis --ms /stimela_mount/msdir/centSplit32k_cal.ms --xaxis channel --yaxis phase --canvas-height 300 --canvas-width 300 --cmap blues --cols 5 --corr 0 --data-column CORRECTED_DATA --field 1 --htmlname /stimela_mount/output/first-centSplit32k-primary_only-bpcal-J0408-6545-phase_chan-CorrXX --iter-axis scan --mem-limit 8GB --num-cores 8
job complete at 2020-06-03 19:43:00.567620 after 0:20:23.273681
```

And here's the relevant shadems log snippet:

```
2020-06-03 14:54:27 - shadems - INFO - :   corrs/Stokes XX YY I Q
2020-06-03 14:54:27 - shadems - INFO - ------------------------------------------------------
2020-06-03 14:54:27 - shadems - INFO - : Data selected for plotting:
2020-06-03 14:54:27 - shadems - INFO - Antenna(s)       : all
2020-06-03 14:54:27 - shadems - INFO - Baseline(s)      : all
2020-06-03 14:54:27 - shadems - INFO - Field(s)         : J0408-6545
2020-06-03 14:54:27 - shadems - INFO - SPW(s)           : all
2020-06-03 14:54:27 - shadems - INFO - Scan(s)          : all
2020-06-03 14:54:27 - shadems - INFO - Channels         : all
2020-06-03 14:54:27 - shadems - INFO - Corr/Stokes      : XX
2020-06-03 14:54:27 - shadems - INFO - ------------------------------------------------------
2020-06-03 14:54:27 - shadems - INFO - axis: FREQ, range (None, None), discretization None
2020-06-03 14:54:27 - shadems - INFO - axis: phase(CORRECTED_DATA), corr None, range (None, None), discretization None
2020-06-03 14:54:27 - shadems - INFO - axis: _(ANTENNA1), corr False, range (None, None), discretization 59
2020-06-03 14:54:27 - shadems - INFO -                  : you have asked for 1 plots employing 3 unique datums
2020-06-03 14:54:27 - shadems - INFO - : Indexing MS and building dataframes (499612 rows, chunk size is 5000)
2020-06-03 14:54:27 - shadems - INFO - : complete
2020-06-03 14:54:27 - shadems - INFO - : rendering 1 dataframes with 2e+09 points into 1 plot types
2020-06-03 14:54:27 - shadems - INFO - : rendering /stimela_mount/output/first-centSplit32k-primary_only-gcal-J0408-6545-CORRECTED_DATA-XX-phase-FREQ-ANTENNA1
.png
2020-06-03 14:54:27 - shadems - INFO - : scanning axis min/max for _FREQ_False CORRECTED_DATA_phase_None ANTENNA1___False
2020-06-03 14:57:38 - shadems - INFO - : rendering using 58 colors (values m000 m001 m002 m003 m004 m006 m007 m008 m009 m010 m011 m012 m013 m014 m015 m016 m01
7 m018 m019 m020 m021 m022 m024 m026 m027 m028 m029 m030 m031 m032 m033 m034 m035 m036 m038 m039 m040 m041 m042 m044 m045 m046 m047 m048 m049 m050 m051 m052 m
053 m054 m055 m056 m057 m058 m059 m060 m061 m062)
2020-06-03 14:57:41 - shadems - INFO -                  : wrote /stimela_mount/output/first-centSplit32k-primary_only-gcal-J0408-6545-CORRECTED_DATA-XX-phase-
FREQ-ANTENNA1.png
2020-06-03 14:57:41 - shadems - INFO - Total time       : 195.2 seconds
2020-06-03 14:57:41 - shadems - INFO - Finished
```

They look consistent to me (e.g. both use corrected column), so might be related to the ragavi version indeed?

And yeah, @o-smirnov, colouring for antennas made it look like there's a uniform distribution of phases between -35 - 35 deg in all channels, but in our defence the colouring was meant to show if the crazy looking ragavi plot is due to some bad antennas. :) I was a bit surprised when I found out they don't even remotely look similar.

**Comment by KshitijT** [comment]
I can confirm this. 

Ragavi made this plot:
![image](https://user-images.githubusercontent.com/9251962/84715636-54f92200-af71-11ea-9e0a-cb904fde4b30.png)

A typical shadems plot for the same field looks like (yes, I went through all the 43 unflagged antenna plots):
![plot-1532022061_8s_cal-J1331+3030-m010-CORRECTED_DATA-XX-YY-amp-phase](https://user-images.githubusercontent.com/9251962/84715729-a43f5280-af71-11ea-8dde-aafe1a3aab5b.png)

The first plot gave me a couple of sleepless nights before I remembered this issue.  Will confirm if this is a ragavi version problem tomorrow.
